ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParseProxy.java

```diff
@@ -0,0 +1,99 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package com.mozilla.telemetry.decoder;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.mozilla.telemetry.transforms.PubsubConstraints;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.metrics.Counter;
+import org.apache.beam.sdk.metrics.Metrics;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.values.PCollection;
+
+public class ParseProxy extends PTransform<PCollection<PubsubMessage>, PCollection<PubsubMessage>> {
+
+  public static ParseProxy of() {
+    return INSTANCE;
+  }
+
+  /////////
+
+  private static final SimpleDateFormat TIMESTAMP_FORMAT = new SimpleDateFormat(
+      "yyyy-MM-dd'T'hh:mm:ss.SSSSSS'Z'");
+  private static final String PROXY_TIMESTAMP = "proxy_timestamp";
+  private static final String SUBMISSION_TIMESTAMP = "submission_timestamp";
+  private static final String X_FORWARDED_FOR = "x_forwarded_for";
+  private static final String X_PIPELINE_PROXY = "x_pipeline_proxy";
+
+  private static final Fn FN = new Fn();
+  private static final ParseProxy INSTANCE = new ParseProxy();
+
+  @VisibleForTesting
+  public static class Fn extends SimpleFunction<PubsubMessage, PubsubMessage> {
+
+    private final Counter countPipelineProxy = Metrics.counter(Fn.class, "count_pipeline_proxy");
```

Comment:
```suggestion
    private final Counter countPipelineProxy = Metrics.counter(Fn.class, "pipeline_proxy");
```

I don't know why I included the "count_" prefix on this in the first place.